tarantool_test.go

```diff
@@ -3972,6 +3972,95 @@ func TestConnect_context_cancel(t *testing.T) {
 	}
 }
 
+func TestConnectIsBlocked(t *testing.T) {
+	const server = "127.0.0.1:3015"
+
+	testDialer := dialer
+	testDialer.Address = server
+
+	ctx, cancel := context.WithTimeout(context.TODO(), 10*time.Second)
+	defer cancel()
+	delay := 2 * time.Second
```

Comment:
Please try to test the code without timers and timeouts. You can implement your own Dialer mock for example.

Reply:
The `retry` logic is implemented in `connection.go`, so I don't see a way to remove `Sleep()`. But I removed the `too early` check. Also, since we don't need a properly functioning connection, we can just make our own dialer that returns the connection after some time. Should I do that? I think it would look a bit nicer, but the test would be more complicated.

Reply:
You could implement a tarantool.Dialer mock and register attempts to call Dialer method. This way you can understand how many attempts there were and the time intervals between them.

Reply:
Thanks, fixed.